.github/workflows/main.yml

```diff
@@ -0,0 +1,98 @@
+name: CI
+
+on: push
+
+jobs:
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - id: component
+        uses: actions-rs/components-nightly@v1
+        with:
+          component: clippy
+      - run: rustup component add clippy
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-targets --all-features -- -D warnings
+
+  format:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - id: component
+        uses: actions-rs/components-nightly@v1
+        with:
+          component: rustfmt
+      - name: Install rustfmt
+        run: rustup component add rustfmt
+
+      - name: Install Erlang/Elixir
+        uses: actions/setup-elixir@v1.0.0
+        with:
+          otp-version: 22.1.3
+          elixir-version: 1.9.2
+
+      - name: Check cargo fmt
+        uses: actions-rs/cargo@v1
+        with:
+          command: fmt
+          args: --all -- --check
+
+      - name: Check mix format (rustler_mix)
+        working-directory: rustler_mix
+        run: mix format --check-formatted
+
+      - name: Check mix format (rustler_tests)
+        working-directory: rustler_tests
+        run: mix format --check-formatted
+
+  test:
+    name: OTP ${{matrix.otp}} / Elixir ${{matrix.elixir}} / Rust ${{matrix.rust}}
+    runs-on: ubuntu-latest
```

Comment:
I wonder, is it possible to run this on macos and windows as well?

Reply:
Ah, found my answer [here](https://github.com/actions/setup-elixir/blob/master/src/installer.js#L34):

> `'@actions/setup-elixir only supports Ubuntu Linux at this time'`